go.mod

```diff
@@ -1,6 +1,8 @@
 module connectrpc.com/conformance
 
-go 1.20
+go 1.21
+
+toolchain go1.21.4
```

Comment:
What's this about? If this is needed for the latest quic-go, we may be best off just closing this PR and maybe consider an upgrade in the future after we've intentionally moved to Go 1.21. (We're currently still supporting 1.20 in this repo, and the Go 1.20 CI job is failing because of this change.)